packages/types/src/metadata/PortableRegistry/PortableRegistry.ts

```diff
@@ -202,25 +205,56 @@ function removeDuplicateNames (lookup: PortableRegistry, names: [number, string
           rewrite[index] = name;
         }
 
-        return [lookupIndex, name, params];
+        return [lookupIndex, name, params, aliases];
       }
 
       return null;
     })
-    .filter((n): n is [number, string, SiTypeParameter[]] => !!n)
-    .map(([lookupIndex, name, params]) => [
+    .filter((n): n is Named => !!n)
+    .map(([lookupIndex, name, params, aliases]) => [
       lookupIndex,
       rewrite[lookupIndex] || name,
-      params
+      params,
+      aliases
     ]);
 }
 
-function extractName (types: PortableType[], { id, type: { params, path } }: PortableType): [number, string, SiTypeParameter[]] | null {
+function removeDuplicateAliases (list: Named[]): Named[] {
+  const dupes: string[] = [];
+
+  for (let i = 0; i < list.length; i++) {
+    const [,,, aliases] = list[i];
+
+    for (let j = 0; j < list.length; j++) {
+      if (i !== j) {
+        const [,,, others] = list[j];
+
+        for (let a = 1; a < aliases.length; a++) {
+          const alias = aliases[a];
+
+          if (!dupes.includes(alias) && others.includes(alias)) {
+            dupes.push(alias);
+          }
+        }
+      }
+    }
+  }
+
+  return list.map(([lookupIndex, name, params, aliases]) => [
+    lookupIndex,
+    name,
+    params,
+    aliases.slice(1).filter((a) => !dupes.includes(a))
+  ]);
```

Comment:
Less memory consuming by mutate array items:
```suggestion
  return list.map((item) => {
    item[3] = item[3].slice(1).filter((a) => !dupes.includes(a))
    return item
  });
```